crates/matrix-store-key/src/lib.rs

```diff
@@ -0,0 +1,510 @@
+// Copyright 2022 The Matrix.org Foundation C.I.C.
+// Copyright 2021 Damir Jelić
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![doc = include_str!("../README.md")]
+#![warn(missing_debug_implementations, missing_docs)]
+
+use blake3::{derive_key, Hash};
```

Comment:
so, this isn't a move from the existing code? And more importantly, it doesn't remove the code from base nor the references on the dependents? 

Did the code actually change? Do I have to review the stuff?

Reply:
As mentioned in chat, this is mostly a copy from the store key in the state store. It changes things a bit, mainly to allow us to use blake3 to hide our keys, values are still encrypted using xchacha20.